.github/workflows/cli.yaml

```diff
@@ -0,0 +1,92 @@
+          container_ip=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "host-control-plane")
+          kubectl config set-cluster "kind-host" --server="https://${container_ip}:6443" --kubeconfig=${HOME}/.kube/host.config
+
+          # load image to host plane
+          kind load docker-image karmada/karmada-aggregated-apiserver:ci-e2e --name host
+          kind load docker-image karmada/karmada-controller-manager:ci-e2e --name host
+          kind load docker-image karmada/karmada-scheduler:ci-e2e --name host
+          kind load docker-image karmada/karmada-webhook:ci-e2e --name host
+
+          # init Karmada control plane
+          karmadactl init --kubeconfig=${HOME}/.kube/host.config \
+            --karmada-data=${HOME}/karmada \
+            --karmada-pki=${HOME}/karmada/pki \
+            --crds='./crds.tar.gz' \
+            --karmada-aggregated-apiserver-image='karmada/karmada-aggregated-apiserver:ci-e2e' \
+            --karmada-controller-manager-image='karmada/karmada-controller-manager:ci-e2e' \
+            --karmada-scheduler-image='karmada/karmada-scheduler:ci-e2e' \
+            --karmada-webhook-image='karmada/karmada-webhook:ci-e2e'
+
+          # create member cluster
+          kind create cluster --name member1 --kubeconfig=${HOME}/.kube/member1.config --image="${CLUSTER_VERSION}" 
+          kind create cluster --name member2 --kubeconfig=${HOME}/.kube/member2.config --image="${CLUSTER_VERSION}" 
+          kubectl wait --for=condition=Ready nodes --all --timeout=600s --kubeconfig=${HOME}/.kube/member1.config
+          kubectl wait --for=condition=Ready nodes --all --timeout=600s --kubeconfig=${HOME}/.kube/member2.config
+
+          # replace ip in kubeconfig
+          container_ip=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "member1-control-plane")
+          kubectl config set-cluster "kind-member1" --server="https://${container_ip}:6443" --kubeconfig=${HOME}/.kube/member1.config
+          container_ip=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "member2-control-plane")
+          kubectl config set-cluster "kind-member2" --server="https://${container_ip}:6443" --kubeconfig=${HOME}/.kube/member2.config
+
+          # join cluster
+          karmadactl --kubeconfig ${HOME}/karmada/karmada-apiserver.config  join member1 --cluster-kubeconfig=${HOME}/.kube/member1.config
+          karmadactl --kubeconfig ${HOME}/karmada/karmada-apiserver.config  join member2 --cluster-kubeconfig=${HOME}/.kube/member2.config
+          kubectl wait --for=condition=Ready clusters --all --timeout=600s  --kubeconfig=${HOME}/karmada/karmada-apiserver.config
+
+          # run a single e2e
+          export KUBECONFIG=${HOME}/karmada/karmada-apiserver.config
+          GO111MODULE=on go install github.com/onsi/ginkgo/v2/ginkgo
+          ginkgo -v --race --trace -p  --focus="Deployment propagation testing"  ./test/e2e/
```

Comment:
Do we need to put all the commands into one script, like `hack/local-up-karmada-with-cli.sh`? @RainbowMango

Reply:
cc @RainbowMango for checking